Source/Diagnostics/ReducedDiags/ParticleHistogram.cpp

```diff
@@ -139,18 +150,27 @@ void ParticleHistogram::ComputeDiags (int step)
     // get parser
     HostDeviceParser<m_nvars> fun_partparser = getParser(m_parser);
 
+    // get filter parser
+    ParserFilter fun_filterparser(m_do_parser_filter, getParser(m_parser_filter), myspc.getMass());
+
     // declare local variables
     Real const bin_min  = m_bin_min;
     Real const bin_size = m_bin_size;
     const bool is_unity_particle_weight =
         (m_norm == NormalizationType::unity_particle_weight) ? true : false;
 
+    // dummy engine required for the filter parser
+    const auto dummy_engine = amrex::RandomEngine{};
```

Comment:
I'm not 100% sure if this is the most correct way to use the RandomEngine.

Reply:
cc @WeiqunZhang for double-checking

Reply:
It's correct.  But I suggest remove the line above and in the kernel do `if (!fun_filterparser(p, amrex::RandomEngine())) {return 0.0_rt;}`.

Reply:
Thanks, I've done that in the latest commit! :+1: